### PR TITLE
Clear up docker readme for dev onboarding

### PR DESCRIPTION
**Pull request by millerdev**
@amstone326 cc @kaapstorm

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -7,11 +7,14 @@ Initial setup
...
-   * If not using the Quick Start terminal, run `eval $(docker-machine env default)` to set up Docker's environment variables.
+   * Preferred: install [Docker for Mac](https://docs.docker.com/docker-for-mac/install/).
+   * Alternate for older Macs that do not support Docker for Mac:
+     * Install [Docker Toolbox](https://docs.docker.com/toolbox/toolbox_install_mac/). Go through the full tutorial, which will create a default machine.
+     * To create a new VM manually, run `docker-machine create default --driver=virtualbox` (not necessary if you followed the Docker Toolbox tutorial).
+     * If not using the Quick Start terminal, run `eval $(docker-machine env default)` to set up Docker's environment variables.
+
 * If you have any HQ services currently running (couch, postgres, redis, etc.), you should stop them now. 
-* Bootstrap the setup:
+* Bootstrap the setup. Skip this step if you came here from the [CommCare HQ README](https://github.com/dimagi/commcare-hq/blob/master/README.md#setup-localsettings).
```
Comment (amstone326) [inline docker/README.md]: Maybe just clarify this and say `Skip this step and go to 'Configure your localsettings' below`

File: docker/README.md [hunk 2]
```diff
@@ -7,11 +7,14 @@ Initial setup
    * Install [Docker](https://docs.docker.com/engine/installation/)
    * Install [Docker Compose](https://docs.docker.com/compose/install/) (Note you can also install in a virtualenv with `$ pip install docker-compose`)
 * OS X
-   * Install [Docker Toolbox](https://docs.docker.com/toolbox/toolbox_install_mac/). Go through the full tutorial, which will create a default machine.
-   * To create a new VM manually, run `docker-machine create default --driver=virtualbox` (not necessary if you followed the Docker Toolbox tutorial).
-   * If not using the Quick Start terminal, run `eval $(docker-machine env default)` to set up Docker's environment variables.
+   * Preferred: install [Docker for Mac](https://docs.docker.com/docker-for-mac/install/).
+   * Alternate for older Macs that do not support Docker for Mac:
+     * Install [Docker Toolbox](https://docs.docker.com/toolbox/toolbox_install_mac/). Go through the full tutorial, which will create a default machine.
+     * To create a new VM manually, run `docker-machine create default --driver=virtualbox` (not necessary if you followed the Docker Toolbox tutorial).
```
Comment (amstone326) [inline docker/README.md]: Is it intentional that these 2 steps (`To create a new VM manually...` and `If not using the Quick Start terminal...`) are sub-steps of using Docker Toolbox? i.e. are they in fact only needed when running Docker Toolbox, not Docker for Mac (I did these 2 steps and they ran fine/didn't do any harm, but they could very well have been unnecessary)

File: docker/README.md [hunk 3]
```diff
@@ -39,14 +42,15 @@ Initial setup
 
 There are two different localsettings configurations, depending on whether HQ is running inside a docker container or on your local machine.
```
Comment (amstone326) [inline docker/README.md]: Just for the sake of people understanding what's going on here, maybe add something at the end of this line like "If you are planning on doing local development, it is recommended to run HQ on your local machine, and use docker only for supporting services"

File: docker/README.md (hunk 2 above)
Comment (millerdev) [reply docker/README.md]: they are only necessary for Docker Toolbox. `docker-machine` is part of Docker Toolbox, and there is no equivalent (that I know of) for Docker for Mac

**Review by kaapstorm — APPROVED** [review]
Nice.

**Review by amstone326 — APPROVED** [review]
Thanks for this, Daniel!

**Pull request merged**